src/source/app/kvsapp.c

```diff
@@ -893,6 +902,26 @@ static int prvPutMediaDoWorkSendEndOfFrames(KvsApp_t *pKvs)
     return res;
 }
 
+static int setupTagsForSession(KvsApp_t *pKvs)
+{
+    int res = KVS_ERRNO_NONE;
+    const size_t tagsListLen = MAX_TAG_AMOUNT - 1; // Note: end of fragment tag counts towards the 10 limit
+    MkvTag_t *tags = kvsMalloc(tagsListLen * sizeof(MkvTag_t));
```

Comment:
Should add a `tags == null ? KVS_ERROR_OUT_OF_MEMORY` check in case out of memory; `kvsMalloc` does not check this, just calls `malloc`.

Reply:
That is a great observation, will add.